huaweicloud/services/ga/resource_huaweicloud_ga_address_group.go

```
@@ -0,0 +1,427 @@
+		KeepResponseBody: true,
+		OkCodes: []int{
+			204,
+		},
+	}
+
+	_, err = deleteIpAddressGroupClient.Request("DELETE", deleteIpAddressGroupPath, &deleteIpAddressGroupOpt)
+	if err != nil {
+		return diag.Errorf("error deleting IP address group: %s", err)
+	}
+
+	stateConf := &resource.StateChangeConf{
+		Pending:      []string{"PENDING"},
+		Target:       []string{"DELETED"},
+		Refresh:      waitIpAddressGroupStatusRefreshFunc(d, meta, true),
+		Timeout:      d.Timeout(schema.TimeoutDelete),
+		Delay:        10 * time.Second,
+		PollInterval: 10 * time.Second,
+	}
+
+	_, err = stateConf.WaitForStateContext(ctx)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	return nil
+}
+
+func createIpAddressGroupWaitingForStateCompleted(ctx context.Context, d *schema.ResourceData, meta interface{}, t time.Duration) error {
+	stateConf := &resource.StateChangeConf{
+		Pending:      []string{"PENDING"},
+		Target:       []string{"COMPLETED"},
+		Refresh:      waitIpAddressGroupStatusRefreshFunc(d, meta, false),
+		Timeout:      t,
+		Delay:        10 * time.Second,
+		PollInterval: 10 * time.Second,
+	}
+	_, err := stateConf.WaitForStateContext(ctx)
+	return err
+}
+
+func waitIpAddressGroupStatusRefreshFunc(d *schema.ResourceData, meta interface{}, isDelete bool) resource.StateRefreshFunc {
+	return func() (interface{}, string, error) {
+		resp, err := getIpAddressGroupInfo(d, meta)
+		if err != nil {
+			if _, ok := err.(golangsdk.ErrDefault404); ok && isDelete {
+				return resp, "DELETED", nil
+			}
+
+			return nil, "ERROR", err
+		}
+
+		respBody, err := utils.FlattenResponse(resp)
+		if err != nil {
+			return nil, "ERROR", err
+		}
+
+		statusRaw, err := jmespath.Search("ip_group.status", respBody)
+		if err != nil {
+			return nil, "ERROR", fmt.Errorf("error parse %s from response body", statusRaw)
+		}
+
+		status := fmt.Sprintf("%v", statusRaw)
+
+		if utils.StrSliceContains([]string{"ERROR"}, status) {
+			return respBody, status, nil
+		}
```

should return an error, such as:
```
return respBody, "ERROR", fmt.Errorf("unexpected address group status: %s", status)
```

---

done